src/HotChocolate/Core/test/Types.Tests/Utilities/InputObjectCompilerTests.cs

```diff
@@ -0,0 +1,28 @@
+
+using System.Collections.Generic;
+using HotChocolate.Types;
+using Snapshooter.Xunit;
+using Xunit;
+
+namespace HotChocolate.Utilities
+{
+    public class InputObjectCompilerTests
+    {
+        [Fact]
+        public void Convert_ObjectWithPropertiesDifferingByCase()
+        {
+            void Action() => SchemaBuilder.New()
+                .AddType<InputObjectType<Foo>>()
+                .Create();
+
+            Assert.Throws<SchemaException>(Action);
+        }
+
+        private class Foo
```

Comment:
This is an issue in GraphQL and there should be no error. Think about people introducing a field with a invalid casing. What you do in GraphQL to fix this is actually introduce the same field with the correct casing and deprecating the old one. They are now side by side and old and new clients work. New clients will always now pic the corrected fields while old clients still work against the old version.